### PR TITLE
Closes #676, adds a signal to create a congifuration for new journals…

### LINKED ISSUE #676: New Journal Error
RelatedObjectDoesNotExist at /submit/start/
Journal has no submissionconfiguration.

### PR DESCRIPTION
**Pull request by ajrbyers**
… and handles those that are missing by creating defaults

### CODE REVIEW

File: src/submission/forms.py [hunk 1]
```diff
@@ -33,6 +33,11 @@ def __init__(self, *args, **kwargs):
         self.fields['competing_interests'].label = ''
         self.fields['comments_editor'].label = ''
 
+        try:
+            journal.submissionconfiguration
+        except models.SubmissionConfiguration.DoesNotExist:
+            models.SubmissionConfiguration.objects.create(journal=journal)
+
```
Comment (mauromsl) [inline src/submission/forms.py]: When do we expect this to happen?

Comment (ajrbyers) [reply src/submission/forms.py]: It happened to me yesterday. Basically when this model was introduced a Signal should have been added to generate them automatically but it wasn't so there may be journals out there without a configuration.

Comment (mauromsl) [reply src/submission/forms.py]: But this code will live here forever, shouldn't we have it on some sort of upgrade script/migration instead?

**Comment by ajrbyers** [comment]
Reverted forms and views code and added a migration to ensure all journals have one. Retained the if created signal.

**Pull request merged**